rlbot/src/state_builder.rs

```diff
@@ -0,0 +1,534 @@
+use rlbot_flat::flat::{
+    DesiredBallState, DesiredCarState, DesiredGameState, DesiredMatchInfo, Rotator, RotatorPartial,
+    Vector3, Vector3Partial,
+};
+use std::ops::Range;
+
+/// Utility for easy construction of [DesiredGameState]s using builder patterns.
+///
+/// Example:
+/// ```example
```

Comment:
make syntax highlighting work in docs:
```suggestion
/// ```rust
```